src/cortex_cli/cortex_cli.py

```diff
@@ -427,6 +433,54 @@ def save_tokens_file(path: str, tokens: dict[str, str], auth_server_url: str) ->
         raise click.ClickException(f'Error writing tokens file, {error}') from error
 
 
+def _validate_cortex_cli_auth(no_auth, config_file) -> Optional[str]:
+    """Checks if provided auth options are correct:
+       - no_auth and config_file are mutually exclusive
+       - if no_auth is not set, config_file must have the CortexCliCommand.default_config_path value
+       - config file must exist and contain a path to an existing tokens_file
+
+    Args:
+        no_auth (bool): --no-auth option value
+        config_file (str): --config-file option value
+    Raises:
+        click.UsageError: if user is not logged in or provided and invalid config
+        click.BadOptionUsage: if both mutually exclusive --no-auth and --config-file are set
+    Returns:
+        str: path to the tokens file if using cortex-cli auth and provided (or default)
+             config-file exists and valid, None if --no-auth is set
+    """
+
+    # --no-auth and --config-file are mutually exclusive
+    if no_auth and config_file:
+        raise click.BadOptionUsage('--no-auth', 'Cannot use both --no-auth and --config-file options.')
+
+    # no config_file to use, no tokens_file
+    if no_auth:
+        return None
+
+    # --config-file was not provided, but has a default value
+    if not config_file:
+        logger.debug('No auth options provided, using default config file: %s', CortexCliCommand.default_config_path)
+        config_file = CortexCliCommand.default_config_path
+
+    # config file, event the default one, should exist
+    if not Path(config_file).is_file():
+        raise click.UsageError('Not logged in. Run `cortex auth login` to log in.')
```

Comment:
In other places, quoting commands or flags is done with single quotes rather than backticks ([example](https://github.com/iqm-finland/cortex-cli/blob/main/src/cortex_cli/cortex_cli.py#L274)). Backticks make more sense, but in any case the choice must be consistent.